### DISCUSSION TITLE
How to render page based on incoming POST request data?

### DISCUSSION THREAD

**Discussion by Irelynx** [nuxt/nuxt#28147] [Questions]
Greetings!

I would like to render page based on incoming data sent to the Nuxt 3 using POST request to specific page only.
Unfortunately, usage of query string parameters is not an option.

Here is an example of what i would like to accomplish:
Some client sends `POST /page/to/render` to Nuxt server with body in JSON format, then Nuxt server respond with "rendered" HTML based on POST request body.

Could someone point me out how i can achieve it?
Thanks in advance!

**Comment by Irelynx** [comment]
I found the following solution, but i am not sure is it optimal or not.

`server/utils/handlePost.ts`
```typescript
import { defineEventHandler, readBody } from 'h3';

export default defineEventHandler(async (event) => {
  if (!event) return;
  if (event.node.req.method === 'POST') {
    const body = await readBody(event);
    event.context.body = body;
  }
});
```

`pages/test/page.vue`
```vue
<script setup lang="ts">
const router = useRouter();
const postData = ref<any>(null);

const event = useRequestEvent();
postData.value = event?.context?.body
</script>

<template>
  <pre>{{ postData }}</pre>
</template>
```

`nuxt.config.ts`
```typescript
export default defineNuxtConfig({
  // ...
  serverHandlers: [
    {
      route: '/test/page',
      middleware: true,
      handler: '~/server/utils/handlePost.ts',
      method: 'POST',
    }
  ],
  // ...
}
```

**Reply by masyoudi** [reply]
no need to place event handler inside `utils` folder and `serverHandlers` in nuxt.config, just put in the event handler inside `routes` folder.


just create file `/server/routes/test/page.post.ts` and this is automatically create post route for you

**Reply by masyoudi** [reply]
you can read the docs in [nitro](https://nitro.unjs.io/guide/routing) website
